File: src/main/java/io/mfedirko/common/validation/ReCaptchaValidator.java
package io.mfedirko.common.validation;

import io.mfedirko.infra.RecaptchaClient;
import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.util.StringUtils;

public class ReCaptchaValidator implements ConstraintValidator<ValidReCaptcha, String> {
    @Autowired
    private RecaptchaClient recaptchaClient;

    @Override
    public boolean isValid(String recaptcha, ConstraintValidatorContext constraintValidatorContext) {
        if (!StringUtils.hasText(recaptcha)) {
            return false;
        }
        return recaptchaClient.isValidCaptcha(recaptcha);
    }
}

File: src/main/kotlin/io/mfedirko/common/validation/ReCaptchaValidator.kt
package io.mfedirko.common.validation

import io.mfedirko.common.infra.RecaptchaClient
import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext
import org.springframework.beans.factory.annotation.Autowired

class ReCaptchaValidator : ConstraintValidator<ValidReCaptcha, String?> {
    @Autowired
    private lateinit var recaptchaClient: RecaptchaClient

    override fun isValid(recaptcha: String?, constraintValidatorContext: ConstraintValidatorContext): Boolean {
        return if (recaptcha.isNullOrBlank()) false
        else recaptchaClient.isValidCaptcha(recaptcha)
    }
}